### ISSUE TITLE
"temporalTypesAsUserData = false" option has no effect?

### ISSUE THREAD

**Issue by usonianhorizon** [LebJe/toml.lua#8]
I have been trying to use this option in my code with no effect to emitted output.  Finally, I just copied and pasted the example shown in the `README.md` file, and again there seems to be no difference in emitted output by inspect.

If I select an individual (date) item to simply print then the output is the string equivalent of the date.

Am I missing something or is the `temporalTypesAsUserData` a NOOP?

advTHANKSance

**Comment by usonianhorizon** [comment]
Hello.

Sorry for the delayed response; late night in the server room :(

I am running FreeBSD 13.1, with a very basic _lua_ installation to test the libraries I need to build out a medium sized standalone application whose 'transaction' data is all in TOML files.

```
$ luarocks54 install --local lpeg
$ luarocks54 install --local getopt
$ luarocks54 install --local inspect
$ luarocks54 install --local toml
```

I copied the MWE given for the use of `temporalTypesAsUserData`, which I am attaching here in case I might have made a transcription error: [t_toml.zip](https://github.com/LebJe/toml.lua/files/10755966/t_toml.zip)

The output I see is as follows:

```
$ lua54 t_toml.lua
{
  date = <userdata 1>,
  datetime = <userdata 2>,
  time = <userdata 3>
}
1979-05-27
{
  date = <userdata 1>,
  datetime = <userdata 2>,
  time = <userdata 3>
}

```

Thank you again for your assistance.

**Comment by LebJe** [comment]
Sorry, by “installation log” I meant the result of running `luarocks54 install --local toml`. Your code is correct but I suspect luarocks is installing the wrong version.

**Comment by usonianhorizon** [comment]
You were right!

Unfortunately, I didn't expect a long log and so didn't think to redirect it to file --- I am running dwm/st without terminal scrolling.  However the relevant part is at the bottom:

>
> ```
> 
> Removing toml 0.1.1-0...
> Removal successful.
> 
> ```

I hadn't thought that toml had been installed so long before on this computer.  I think we can close the issue.

Again, thank you.

```

In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/src/DataTypes/DateAndTime/dateAndTime.cpp:1:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/src/DataTypes/DateAndTime/dateAndTime.hpp:7:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/sol.hpp:52:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/stack.hpp:32:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/stack_check_get.hpp:28:
/tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/stack_check_get_qualified.hpp:35:32: warning: unknown warning group '-Wmaybe-uninitialized', ignored [-Wunknown-warning-option]
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
                               ^
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/src/DataTypes/DateAndTime/dateAndTime.cpp:1:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/src/DataTypes/DateAndTime/dateAndTime.hpp:7:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/sol.hpp:54:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/function.hpp:28:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/unsafe_function.hpp:30:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/function_result.hpp:27:
/tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/protected_function_result.hpp:104:32: warning: unknown warning group '-Wmaybe-uninitialized', ignored [-Wunknown-warning-option]
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
                               ^
2 warnings generated.
[ 71%] Building CXX object CMakeFiles/toml.lua.dir/src/DataTypes/TOMLInt/TOMLInt.cpp.o
[ 85%] Building CXX object CMakeFiles/toml.lua.dir/src/utilities/utilities.cpp.o
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/src/utilities/utilities.cpp:1:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/src/utilities/utilities.hpp:7:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/sol.hpp:52:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/stack.hpp:32:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/stack_check_get.hpp:28:
/tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/stack_check_get_qualified.hpp:35:32: warning: unknown warning group '-Wmaybe-uninitialized', ignored [-Wunknown-warning-option]
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
                               ^
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/src/utilities/utilities.cpp:1:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/src/utilities/utilities.hpp:7:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/sol.hpp:54:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/function.hpp:28:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/unsafe_function.hpp:30:
In file included from /tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/function_result.hpp:27:
/tmp/luarocks_toml-0.2.0-0-sS26zu/toml.lua/build.luarocks/_deps/sol2-src/include/sol/protected_function_result.hpp:104:32: warning: unknown warning group '-Wmaybe-uninitialized', ignored [-Wunknown-warning-option]
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
                               ^
2 warnings generated.
[100%] Linking CXX shared module toml.so
[100%] Built target toml.lua
Consolidate compiler generated dependencies of target toml.lua
[100%] Built target toml.lua
Install the project...
-- Install configuration: "Release"
-- Installing: /home/pavneet/.luarocks/lib/luarocks/rocks-5.4/toml/0.2.0-0/lib/toml.so
toml 0.2.0-0 is now installed in /home/pavneet/.luarocks (license: MIT)

Checking stability of dependencies in the absence of
toml 0.1.1-0...

Removing toml 0.1.1-0...
Removal successful.
```

**Comment by LebJe** [comment]
Awesome, glad it works!
